### ISSUE TITLE
Assuming as equal two classes differentiated by a number at the and 

### ISSUE THREAD

**Issue by adrianmrit** [purifycss/purifycss#147]
I have the following code:
```
var purify = require('purify-css');
var content = '<div class="hello">sad<//div><div class="hello2">asdasda<//div>';
var css = '.hello { color: green; } .hello3 { display: block; }';

  var options = {
      output: 'purified.css',

  // Will minify CSS code in addition to purify.
  minify: true,

  // Logs out removed selectors.
  rejected: true,
  info: true
};

purify(content, css, options);
```
And then the content of purified.css is:
'.hello { color: green; } .hello3 { display: block; }'

It is not deleting hello3, but if I change css as follows, then it works:
`var css = '.hello { color: green; } .h { display: block; }';`
And then the content of purified.css is:
'.hello { color: green; }'
I guess it is not taking into account the number at the end of the class.

**Comment by adrianmrit** [comment]
I just fixed the Issue, hope it will be accepted :)

**Comment by joaovpmamede** [comment]
Same problem here I guess.
Reported [here](https://github.com/webpack-contrib/purifycss-webpack/issues/73#event-1031550061).

**Comment by adamwathan** [comment]
Solving this would be a huge win; I have tons of utility classes ending in numbers or containing non-standard characters like `:`, for example:

`.md\:mb-5`

...used like:

`<div class="md:mb-5">...</div>`

...which adds a bottom margin of my 5th spacing unit on medium screens and larger.

Currently, after running PurifyCSS, Chrome still reports 97% of my CSS as being unused:

![image](https://user-images.githubusercontent.com/4323180/28828505-0097ccf6-769f-11e7-83db-3409eb56b279.png)

**Comment by zackkrida** [comment]
Looks like there's a PR here: https://github.com/purifycss/purifycss/pull/148

No momentum in a bit, however...

**Comment by JayHoltslander** [comment]
This is unfortunate. If I never use a grid's ``.col11`` or ``.col1`` for example then why should they be included? I guess purify-css isn't for me.

**Comment by adamwathan** [comment]
@JayHoltslander Try [purgecss](https://github.com/FullHuman/purgecss), I've been using it for close to a year now and it's worked flawlessly.

**Comment by JayHoltslander** [comment]
@adamwathan Yeah was looking at both [purgecss](https://www.purgecss.com/) and [uncss](https://github.com/uncss/uncss). Do you have any feelings on uncss vs purgecss?

**Comment by adamwathan** [comment]
Uncss wouldn’t work for me because it can’t be run against arbitrary
template languages like PHP even. It seems it tries to be a bit too smart
IMO, by actually running client side JS and parsing the DOM to find
classes, etc.

Purgecss is by design very “dumb”, it just iterates over all of your CSS
classes and looks for them as simple string matches in any source file you
provide, like a template or JS file. It doesn’t actually parse HTML or
anything, it literally just runs a regex against the files you provide. In
my experience that makes it very very predictable and easy to use because
it’s obvious how to author your templates in a “purgeable” way because the
mental model is so straightforward.
